test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentLinkedQueueTest.scala

```diff
@@ -0,0 +1,174 @@
+      expect(pq.add(-0.0)).toBeTruthy
+      expect(pq.add(Double.NaN)).toBeTruthy
+
+      expect(pq.poll.equals(1.0)).toBeTruthy
+
+      expect(pq.poll.equals(+0.0)).toBeTruthy
+
+      expect(pq.poll.equals(-0.0)).toBeTruthy
+
+      expect(pq.peek.isNaN).toBeTruthy
+
+      expect(pq.remove(Double.NaN)).toBeTruthy
+
+      expect(pq.isEmpty).toBeTruthy
+    }
+
+    it("could be instantiated with a prepopulated Collection") {
+      val l = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val pq = new ConcurrentLinkedQueue[Int](l)
+
+      expect(pq.size()).toEqual(5)
+      for (i <- l) {
+        expect(pq.poll()).toEqual(i)
+      }
+      expect(pq.isEmpty).toBeTruthy
+    }
+
+    it("should be cleared in a single operation") {
+      val l = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val pq = new ConcurrentLinkedQueue[Int](l)
+
+      expect(pq.size()).toEqual(5)
+      pq.clear()
+      expect(pq.size()).toEqual(0)
+    }
+
+    it("should add multiple elemnt in one operation") {
+      val l = asJavaCollection(Set(1, 5, 2, 3, 4))
+      val pq = new ConcurrentLinkedQueue[Int]()
+
+      expect(pq.size()).toEqual(0)
+      pq.addAll(l)
+      expect(pq.size()).toEqual(5)
+      pq.add(6)
+      expect(pq.size()).toEqual(6)
+    }
+
+    it("should check contained values even in double corner cases") {
+      val pq = new ConcurrentLinkedQueue[Double]()
+
+      expect(pq.add(11111.0)).toBeTruthy
+      expect(pq.size()).toEqual(1)
+      expect(pq.contains(11111.0)).toBeTruthy
+      expect(pq.iterator.next()).toEqual(11111.0)
+
+      expect(pq.add(Double.NaN)).toBeTruthy
+      expect(pq.size()).toEqual(2)
+      expect(pq.contains(Double.NaN)).toBeTruthy
+      expect(pq.contains(+0.0)).toBeFalsy
+      expect(pq.contains(-0.0)).toBeFalsy
+
+      expect(pq.remove(Double.NaN)).toBeTruthy
+      expect(pq.add(+0.0)).toBeTruthy
+      expect(pq.size()).toEqual(2)
+      expect(pq.contains(Double.NaN)).toBeFalsy
+      expect(pq.contains(+0.0)).toBeTruthy
+      expect(pq.contains(-0.0)).toBeFalsy
+
+      expect(pq.remove(+0.0)).toBeTruthy
+      expect(pq.add(-0.0)).toBeTruthy
+      expect(pq.size()).toEqual(2)
+      expect(pq.contains(Double.NaN)).toBeFalsy
+      expect(pq.contains(+0.0)).toBeFalsy
+      expect(pq.contains(-0.0)).toBeTruthy
+
+      expect(pq.add(+0.0)).toBeTruthy
+      expect(pq.add(Double.NaN)).toBeTruthy
+      expect(pq.contains(Double.NaN)).toBeTruthy
+      expect(pq.contains(+0.0)).toBeTruthy
+      expect(pq.contains(-0.0)).toBeTruthy
+    }
+
+    it("should provide a weakly consistent iterator") {
+      val queue = new ConcurrentLinkedQueue[Int]()
+      queue.add(1)
+      queue.add(2)
+      val iter1 = queue.iterator()
+      expect(iter1.next()).toEqual(1)
+      expect(iter1.hasNext).toBeTruthy
+      queue.remove(2)
+      expect(iter1.hasNext).toBeTruthy // yes, must still be truthy
+      expect(iter1.next()).toEqual(2)
+      expect(iter1.hasNext).toBeFalsy
+
+      queue.clear()
```

Comment:
Create a new `val queue2` here, to make absolutely certain the two tests are independent.